pkgs/development/python-modules/mock-open/default.nix

```diff
@@ -1,17 +1,25 @@
-{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, mock }:
+{ lib, buildPythonPackage, fetchFromGitHub, fetchpatch, pythonOlder, mock }:
 
 buildPythonPackage rec {
   pname = "mock-open";
-  version = "1.3.1";
+  version = "1.4.0";
 
   # no tests in PyPI tarball
   src = fetchFromGitHub {
     owner = "nivbend";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0ikhrhlkl5c7qbigpsv44jw89ws1z7j06gzyg5dh1ki533ifbjm2";
+    sha256 = "0qlz4y8jqxsnmqg03yp9f87rmnjrvmxm5qvm6n1218gm9k5dixbm";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "ascii-only.patch";
+      url = "https://github.com/das-g/mock-open/commit/521ff260da127949fe4aceff1667cba223c5b07b.patch";
+      sha256 = "0ampbhk7kwkn0q5d2h9wrflkr8fji2bybmdck4qdzw1qkslfwwrn";
```

Comment:
This patch (from nivbend/mock-open#10) is only needed on Python 2. Should it be applied unconditionally or only when building for Python 2? (It doesn't do any harm on Python 3.)

Reply:
I'd make py2-only because any patch has the possibility to cause problems that stop us auto-bumping.

Reply:
Changed with https://github.com/NixOS/nixpkgs/compare/f19d0ea6acd1703c85ade318c54d4f157053aa4d..399d7afeb67b660d4691cc8d59fbb949699a39c6